codegenerator/cli/npm/envio/src/LogSelection.res

```diff
@@ -56,10 +49,59 @@ let compressTopicSelections = (topicSelections: array<topicSelection>) => {
 
 type t = {
   addresses: array<Address.t>,
-  topicSelections: array<topicSelection>,
+  topicSelections: array<Internal.topicSelection>,
 }
 
 let make = (~addresses, ~topicSelections) => {
   let topicSelections = compressTopicSelections(topicSelections)
   {addresses, topicSelections}
 }
+
+let fromEventFiltersOrThrow = {
+  let noopGetter = _ => []
+
+  (
+    ~chain,
+    ~eventFilters: option<Js.Json.t>,
+    ~sighash,
+    ~topic1=noopGetter,
+    ~topic2=noopGetter,
+    ~topic3=noopGetter,
+  ) => {
+    let topic0 = [sighash->EvmTypes.Hex.fromStringUnsafe]
+    switch eventFilters->Js.typeof {
+    | "undefined" => [
+        {
```

Comment:
Moved this from the Types.res so it's possible to evaluate it from the source when we know chain and addresses. Also, it reduced dependency on codegen.